src/main/java/com/esri/arcgisruntime/toolkit/skins/FeatureTemplatePickerTilePaneSkin.java

```diff
@@ -0,0 +1,329 @@
+    Orientation orientation = getSkinnable().getOrientation();
+    // keep a reference to the current children
+    List<Node> children = contentPane.getChildren();
+    // create a new content pane based on the orientation
+    contentPane = orientation == Orientation.HORIZONTAL ? new HBox() : new VBox();
+    // add a style class so the picker can be styled via CSS
+    contentPane.getStyleClass().add("feature-template-picker");
+    // add a CSS pseudo class so the picker can be styled based on it orientation
+    contentPane.pseudoClassStateChanged(HORIZONTAL_PSEUDO_CLASS, orientation == Orientation.HORIZONTAL);
+    // copy the children back into the new content pane
+    contentPane.getChildren().addAll(children);
+    scrollPane.setContent(contentPane);
+    // size the content pane to the scroll pane
+    scrollPane.setFitToWidth(orientation == Orientation.VERTICAL);
+    scrollPane.setFitToHeight(orientation == Orientation.HORIZONTAL);
+  }
+
+  /**
+   * Creates a VBox with a label and tile pane representing a feature template group.
+   *
+   * @param featureTemplateGroup feature template group to show
+   * @return VBox
+   */
+  private VBox createVBoxForFeatureTemplateGroup(FeatureTemplateGroup featureTemplateGroup) {
+    VBox vBox = new VBox();
+    // add a style class so the group can be styled via CSS
+    vBox.getStyleClass().add("feature-template-group");
+    vBox.setVisible(false); // wait until the group's feature layer is loaded before displaying
+    vBox.managedProperty().bind(vBox.visibleProperty()); // hide the view when not visible
+
+    // add a label to the view to show the group's feature layer's name
+    Label label = new Label();
+    label.managedProperty().bind(label.visibleProperty()); // hide the label when not visible
+    vBox.getChildren().add(label);
+
+    // add a tile pane to the view to display each feature template item in the group
+    TilePane tilePane = new TilePane();
+    tilePane.getStyleClass().add("tile-pane");
+    // the tile pane's orientation should be the opposite of the control's orientation
+    tilePane.setOrientation(getSkinnable().getOrientation() == Orientation.HORIZONTAL ? Orientation.VERTICAL :
+            Orientation.HORIZONTAL);
+    getSkinnable().orientationProperty().addListener((observable, oldValue, newValue) ->
+      tilePane.setOrientation(newValue == Orientation.HORIZONTAL ? Orientation.VERTICAL : Orientation.HORIZONTAL)
+    );
+    // switch alignment to center when there is only one row/column, otherwise top-left.
+    tilePane.widthProperty().addListener((observable, oldValue, newValue) -> {
+      if (newValue.doubleValue() < tilePane.getTileWidth() * 2) {
+        tilePane.setAlignment(Pos.TOP_CENTER);
+      } else {
+        tilePane.setAlignment(Pos.TOP_LEFT);
+      }
+    });
+    vBox.getChildren().add(tilePane);
+
+    // ensure the group's feature layer is loaded to get its name for the label and its list of feature templates
```

Comment:
If it isn't loaded does the template picker load it somewhere?

Reply:
It's loaded in FeatureTemplateGroup but it's added to the picker before load, so we need to still make sure it's loaded here. I added a load call to make it obvious it's being loaded.